src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -0,0 +1,1564 @@
+            out bool hasGetAccessor,
+            out bool hasSetAccessor,
+            out bool isInitOnly,
+            out CSharpSyntaxNode? getSyntax,
+            out CSharpSyntaxNode? setSyntax);
+
+        protected abstract void CheckForBlockAndExpressionBody(CSharpSyntaxNode syntax, DiagnosticBag diagnostics);
+
+#nullable restore
+
+        internal sealed override ImmutableArray<string> NotNullMembers =>
+            GetDecodedWellKnownAttributeData()?.NotNullMembers ?? ImmutableArray<string>.Empty;
+
+        internal sealed override ImmutableArray<string> NotNullWhenTrueMembers =>
+            GetDecodedWellKnownAttributeData()?.NotNullWhenTrueMembers ?? ImmutableArray<string>.Empty;
+
+        internal sealed override ImmutableArray<string> NotNullWhenFalseMembers =>
+            GetDecodedWellKnownAttributeData()?.NotNullWhenFalseMembers ?? ImmutableArray<string>.Empty;
+
+        internal bool IsExpressionBodied
+            => (_propertyFlags & Flags.IsExpressionBodied) != 0;
+
+        private void CheckInitializer(
+            bool isAutoProperty,
+            bool isInterface,
+            bool isStatic,
+            Location location,
+            DiagnosticBag diagnostics)
+        {
+            if (isInterface && !isStatic)
+            {
+                diagnostics.Add(ErrorCode.ERR_InstancePropertyInitializerInInterface, location, this);
+            }
+            else if (!isAutoProperty)
+            {
+                diagnostics.Add(ErrorCode.ERR_InitializerOnNonAutoProperty, location, this);
+            }
+        }
+
+        public override RefKind RefKind
+        {
+            get
+            {
+                return _refKind;
+            }
+        }
+
+        public override TypeWithAnnotations TypeWithAnnotations
+        {
+            get
+            {
+                if (_lazyType == null)
+                {
+                    var diagnostics = DiagnosticBag.GetInstance();
+                    var binder = this.CreateBinderForTypeAndParameters();
+                    var result = this.ComputeType(binder, CSharpSyntaxNode, diagnostics);
+                    if (Interlocked.CompareExchange(ref _lazyType, new TypeWithAnnotations.Boxed(result), null) == null)
+                    {
+                        this.AddDeclarationDiagnostics(diagnostics);
+                    }
+                    diagnostics.Free();
+                }
+
+                return _lazyType.Value;
+            }
+        }
+
+        internal bool HasPointerType
+        {
+            get
+            {
+                if (_lazyType != null)
+                {
+
+                    var hasPointerType = _lazyType.Value.DefaultType.IsPointerOrFunctionPointer();
+                    Debug.Assert(hasPointerType == hasPointerTypeSyntactically());
+                    return hasPointerType;
+                }
+
+                return hasPointerTypeSyntactically();
+
+                bool hasPointerTypeSyntactically()
```

Comment:
>bool hasPointerTypeSyntactically() [](start = 16, length = 34)

Consider making this an abstract method as well

Reply:
If we remove the idea that we always have a TypeSyntax, as you suggest below, what would you expect the implementation for this to be. Assert unreachable?

Reply:
Perhaps it makes more sense to make `HasPointerType` abstract in that case, rather than `HasPointerTypeSyntactically`

Reply:
> If we remove the idea that we always have a TypeSyntax, as you suggest below, what would you expect the implementation for this to be. Assert unreachable?

Take a look at [https://github.com/dotnet/roslyn/pull/45424](https://github.com/dotnet/roslyn/pull/45424). I think I had to deal with similar situation there

---
In reply to: [445212683](https://github.com/dotnet/roslyn/pull/45411#discussion_r445212683) [](ancestors = 445212683)

Reply:
> Perhaps it makes more sense to make HasPointerType abstract in that case, rather than HasPointerTypeSyntactically

I think we want to own access to the _lazyType field

---
In reply to: [445212894](https://github.com/dotnet/roslyn/pull/45411#discussion_r445212894) [](ancestors = 445212894)